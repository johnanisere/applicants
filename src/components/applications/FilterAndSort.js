import qs from "qs";
import React from "react";
import _ from "underscore";
import { withRouter } from "react-router-dom";

import SortBar from "../SortBar";
import FilterBar from "../FilterBar";
import Container from "../../blocks/Container";
import { onResolveUrl } from "../../utils/helpers";
import sort, { sortParams } from "./constants/sort";

class FilterAndSort extends React.Component {
  state = {
    sortby: ""
  };

  onHandleChange = ({ target: { value, name } }) => {
    this.setState({ [name]: value }, () =>
      this.resolveUrl("sort", sortParams[value])
    );
  };

  resolveUrl = (name, value) => {
    const {
      location: { search, pathname },
      push
    } = this.props.history;

    onResolveUrl(qs, search, name, value, push, pathname);
  };

  onGetInitialUrl = () => {
    const { search } = this.props.history.location;
    const { sort } = qs.parse(search, {
      ignoreQueryPrefix: true
    });
    if (sort) {
      this.setState({ sortby: _.invert(sortParams)[sort] });
    }
  };

  componentDidMount() {
    this.onGetInitialUrl();
  }

  render() {
    const { sortby } = this.state;
    return (
      <Container>
        <FilterBar resolveUrl={this.resolveUrl} />
        <SortBar
          list={sort}
          name="sortby"
          value={sortby}
          onHandleChange={this.onHandleChange}
        />
      </Container>
    );
  }
}

export default withRouter(FilterAndSort);
