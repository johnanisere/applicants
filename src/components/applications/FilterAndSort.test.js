import React from "react";
import { shallow } from "enzyme";

import FilterAndSort from "./FilterAndSort";

it("FilterAndSort component exists", () => {
  const wrapper = shallow(<FilterAndSort />);
  expect(wrapper.exists()).toBe(true);
});

it("FilterAndSort component renders", () => {
  const wrapper = shallow(<FilterAndSort />);
  expect(wrapper).toMatchSnapshot();
});
