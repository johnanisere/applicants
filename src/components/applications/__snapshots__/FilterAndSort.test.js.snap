// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterAndSort component renders 1`] = `ShallowWrapper {}`;
